socorro/external/boto/crashstorage.py

```diff
@@ -177,6 +178,8 @@ def build_s3_dirs(prefix, name_of_thing, crash_id):
     #--------------------------------------------------------------------------
     @staticmethod
     def do_save_raw_crash(boto_s3_store, raw_crash, dumps, crash_id):
+        if dumps is None:
+            dumps = MemoryDumpsMapping()
```

Comment:
no, because it is perfectly legitimate for a FileDumpsMapping to have been passed in at that time.  If something other than those two were passed and that thing did not have a ```as_memory_dumps_mapping``` method, then effectively, line 203 would be doing that assertion for us.   Perhaps we could assert that whatever is passed in must have a ```as_memory_dumps_mapping``` attribute.